lua/lazyvim/plugins/extras/test/core.lua

```diff
@@ -0,0 +1,38 @@
+return {
+  {
+    "folke/which-key.nvim",
+    optional = true,
+    opts = {
+      defaults = {
+        ["<leader>t"] = { name = "+test" },
+      },
+    },
+  },
+  {
+    "nvim-neotest/neotest",
+    config = function(_, opts)
+      local neotest_ns = vim.api.nvim_create_namespace("neotest")
+      vim.diagnostic.config({
+        virtual_text = {
+          format = function(diagnostic)
+            -- Replace newline and tab characters with space for more compact diagnostics
+            local message = diagnostic.message:gsub("\n", " "):gsub("\t", " "):gsub("%s+", " "):gsub("^%s+", "")
+            return message
+          end,
+        },
+      }, neotest_ns)
+
+      require("neotest").setup(opts)
+    end,
+    -- stylua: ignore
+    keys = {
+      { "<leader>tr", function() require("neotest").run.run() end, desc = "Run Nearest" },
+      { "<leader>tR", function() require("neotest").run.run(vim.fn.expand("%")) end, desc = "Run File" },
+      { "<leader>td", function() require("neotest").run.run({strategy = "dap"}) end, desc = "Debug Nearest" },
```

Comment:
@folke How can we handle this? It should probably only be enabled when `nvim-dap` is present.

Reply:
Take it outside of the neotest spec and add a new spec for `nvim-dap`. Add the keys there. And make that spec optional.

`keys` are always merged even as a list, so no need to use a keys function for that

Reply:
Perfect solution! This `optional` property is a great addition 👍

Reply:
Yeah, should have added that a while ago :) Useful for other LazyVim plugin specs too. Making some changes there as well